### PR TITLE
Lancedb

### PR DESCRIPTION
**Pull request by PrashantDixit0**
**Please describe the purpose of this pull request.**
This is an integration with LanceDB

**How to test**
I have written `tests/test_local_archival.py` and `tests/test_storage.py` and they can be tested in the same way as `Postgres`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -47,6 +47,7 @@ pg8000 = {version = "^1.30.3", optional = true}
 torch = {version = ">=2.0.0, !=2.0.1, !=2.1.0", optional = true}
 websockets = "^12.0"
 docstring-parser = "^0.15"
+lancedb = "^0.3.3"
```
Comment (sarahwooders) [inline pyproject.toml:50]: Could you please add this an an option dependency, and make an extra called lancedb? e.g. you you can do `pip install pymemgpt[lancedb]` -- we are trying to keep the dependencies are minimal as possible

**Review by sarahwooders — CHANGES_REQUESTED** [review]
Could you please add documentation in this page https://memgpt.readthedocs.io/en/latest/storage/ (I think `docs/storage.md`) for LanceDB that includes installation/setup instructions for LanceDB (similar to the Postgres section)? 

The code overall looks good (thanks for the contribution!) -- I'll try to run some testing sometime tomorrow.

**Comment by sarahwooders** [comment]
Also one more note - I saw you updated the `pyproject.toml`, but if you could also update the dependencies through poetry (so the poetry.lock) is updated that would be great.

`pip install poetry`
`poetry add lancedb` 

You can also fix the formatting with `poetry run black . -l 140`

**Comment by PrashantDixit0** [comment]
Yes Sure, I'll do that

**Comment by PrashantDixit0** [comment]
Done @sarahwooders added docs and dependencies with updated poetry.lock

File: pyproject.toml (hunk 1 above)
Comment (sarahwooders) [reply pyproject.toml:50]: Could you please change this to `lancedb = {version = "^0.3.3", optional = true}`?

File: memgpt/autogen/examples/memgpt_coder_autogen.ipynb [hunk 2]
```diff
@@ -38,7 +38,8 @@
    "outputs": [],
    "source": [
     "import openai\n",
-    "openai.api_key=\"YOUR_API_KEY\""
+    "\n",
```
Comment (sarahwooders) [inline memgpt/autogen/examples/memgpt_coder_autogen.ipynb:41]: Please remove this diff

**Review by sarahwooders — CHANGES_REQUESTED** [review]
This looks great, thanks for making these changes! I added two more minor comments, but once those changes are made we can merge this in.

File: memgpt/autogen/examples/memgpt_coder_autogen.ipynb (hunk 2 above)
Comment (PrashantDixit0) [reply memgpt/autogen/examples/memgpt_coder_autogen.ipynb:41]: Ok, I'll make these changes asap

**Comment by sarahwooders** [comment]
Could you also fix the formatting on the changed files with `poetry run black . -l 140`? If you give me access to your branch I can also run it for you.

**Review by sarahwooders — APPROVED** [review]
lgtm - thanks for the awesome contribution!

**Comment by PrashantDixit0** [comment]
Thank you @sarahwooders for merging lancedb vectordb

**Pull request merged**